torch/_inductor/utils.py

```diff
@@ -2620,11 +2620,24 @@ def register_op_dtype_propagation_rules(
     )
 
 
+def get_current_backend() -> str:
+    from torch._inductor.virtualized import V
+
+    device_str = V.graph.get_current_device_or_throw().type
+    if device_str == "cpu":
+        return config.cpu_backend
+    elif device_str == "mps":
+        return "mps"
+    else:
+        return config.cuda_backend
+
+
 def upcast_compute_type(dtype: torch.dtype) -> torch.dtype:
     """Maybe upcast [b]float16 to float32"""
-    if config.triton.codegen_upcast_to_fp32 and (
-        dtype in (torch.float16, torch.bfloat16)
-    ):
+    triton_upcast = (
+        get_current_backend() == "triton" and config.triton.codegen_upcast_to_fp32
+    )
```

Comment:
Can we check `dtype in (torch.float16, torch.bfloat16)` first  and then `get_current_backend() == "triton" and config.triton.codegen_upcast_to_fp32` ?  i think all the repeated TLS lookups is whats causing a small slowdown.

Reply:
Thanks eellison. Seems that the UT passed with your suggested changes. I guess that we can land this PR now?